### PR TITLE
feat(MessageBus): Update to lasted go-mod-messaging and create sample profile for MQTT Message Bus

### PR DESCRIPTION
**Pull request by lenny-intel**
Updated to use latest go-mode-messaging and created the new sample-mqtt-msg-bus profile that configures the MessageBus to use MQTT broker.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,4 +2,6 @@ module github.com/edgexfoundry/app-service-configurable
 
 go 1.13
 
-require github.com/edgexfoundry/app-functions-sdk-go v1.0.0
+require github.com/edgexfoundry/app-functions-sdk-go v1.0.1-dev.21
+
+replace github.com/edgexfoundry/app-functions-sdk-go => ../app-functions-sdk-go
```
Comment (rsdmike) [inline go.mod:7]: Whoops -- dont want this here

File: res/sample-mqtt-msg-bus/configuration.toml [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    # If True, the incoming data to the functions pipeline will be []byte, i.e not marshaled to an Event
+    # This is useful when the incoming data isn't an EdgeX Event.
+    UseTargetTypeOfByteArray = true
+
+    # ExecutionOrder specifies which functions to run and the order to run them.
+    # All functions listed must have an entry below in the Pipeline.Functions section
+    ExecutionOrder = "TransformToXML, SetOutputData"
+
+    # The Pipeline.Functions sections define the parameter configuration for each specific function.
+    # These function names must match a function define in the configurable package of the SDK.
```
Comment (rsdmike) [inline res/sample-mqtt-msg-bus/configuration.toml:20]: define => defined

File: res/sample-mqtt-msg-bus/configuration.toml [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+[Database]
+Type = "mongodb"
+Host = "localhost"
+Port = 27017
+Timeout = "30s"
+Username = "appservice"
+Password = "password"
+
+# SecretStore is required when Store and Forward is enabled and running with security
+# so Databse credentails can be pulled from Vault.
```
Comment (rsdmike) [inline res/sample-mqtt-msg-bus/configuration.toml:101]: Databse credentails => database credentials

File: res/sample-mqtt-msg-bus/configuration.toml [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+Type = "mqtt"
+    [MessageBus.SubscribeHost]
+        Host = "localhost"
+        Port = 1883
+        Protocol = "tcp"
+    [MessageBus.PublishHost]
+        Host = "localhost"
+        Port = 1883
+        Protocol = "tcp"
+    [MessageBus.Optional]
```
Comment (rsdmike) [inline res/sample-mqtt-msg-bus/configuration.toml:138]: does the go-mod-messaging library handle "Secrets" of these values from the vault?

File: go.mod (hunk 1 above)
Comment (lenny-intel) [reply go.mod]: oops, forgot to push the update to remove the replace. Fixed.

File: res/sample-mqtt-msg-bus/configuration.toml (hunk 4 above)
Comment (lenny-intel) [reply res/sample-mqtt-msg-bus/configuration.toml:138]: No, we will need to address this during boot strapping.

File: res/sample-mqtt-msg-bus/configuration.toml (hunk 3 above)
Comment (lenny-intel) [reply res/sample-mqtt-msg-bus/configuration.toml:101]: You will see the other sample get these 2 comment fixes when your review...

File: res/sample/configuration.toml [hunk 5]
```diff
@@ -98,7 +98,7 @@ Username = "appservice"
 Password = "password"
 
 # SecretStore is required when Store and Forward is enabled and running with security
-# so Databse credentails can be pulled from Vault.
+# so databse credentails can be pulled from Vault.
```
Comment (mikejohanson) [inline res/sample/configuration.toml:101]: It looks like you almost fixed this :)

Comment (mikejohanson) [reply res/sample/configuration.toml:101]: database and credentials are both spelled wrong

**Pull request merged**